File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
   //Error message for insufficient arguements
   if (argc != 2)
   {
       printf("Usage: ./caesar k\n");
       return 1;
   }

   //Convert key from string to int
   int k = atoi(argv[1]);

   //Error message for negative integer key
   if (k < 0)
   {
       printf("Key must be a non-negative integer.\n");
       return 1;
   }

   //Prompt user for plain text
   printf("plaintext: ");
   string p = get_string();
   printf("ciphertext: ");

   //Loop through characters
   for (int i = 0, n = strlen(p); i < n; i++)
   {
      //Upper and lower case determination and conversion, print non-alpha
      if (islower(p[i]))
      {
         printf("%c", ((((p[i] + k) - 97) % 26) + 97));
      }
      else if (isupper(p[i]))
      {
         printf("%c", ((((p[i] + k) - 65) % 26) + 65));
      }
      else
      {
         printf("%c", p[i]);
      }

   }

   //Finish with new line print
   printf("\n");
   return 0;
}

File: caesar.py
import cs50
import sys

def main():
    #Error message if key isn't provided
    if len(sys.argv) != 2:
        print("Usage: python caesar.py k")
        exit(1)
    #Convert key from string to int
    k = int(sys.argv[1])
    #Error message if negative key provided
    if k < 0:
        print("Key must be a non-negative integer.")
        exit(1)
    
    #Prompt user for plaintext, begin printing ciphertext
    print("plaintext:", end=" ")
    p = cs50.get_string()
    print("ciphertext:", end=" ")
    
    #Upper and lower case determination and conversion, print non-alpha
    for i in range(len(p)):
        if p[i].islower():
            print("{}".format(chr(((((ord(p[i]) + k) - 97) % 26) + 97))), end="")
        elif p[i].isupper():
            print("{}".format(chr(((((ord(p[i]) + k) - 65) % 26) + 65))), end="")
        else: 
            print("{}".format(p[i]), end="")
    #Print new line at end
    print()
        
#Call main function
if __name__ == "__main__":
    main()